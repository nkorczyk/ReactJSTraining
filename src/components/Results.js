import React from 'react';

const Results = () => {
  return (
    <div className="list2 grey lighten-1">
      <h6 className="white-text text-darken-2 grey lighten-1 grey3 left">0 movies found</h6>
      <h6 className="white-text text-darken-2 grey lighten-1 grey3 right">rating</h6>
      <h6 className="white-text text-darken-2 grey lighten-1 grey3 right">release date</h6>
      <h6 className="white-text text-darken-2 grey lighten-1 grey3 right">Sort by</h6>
    </div>
  )
}

export default Results;
