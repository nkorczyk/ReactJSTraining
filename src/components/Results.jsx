import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { sortMovies } from '../actions/actionCreator';
import CONSTANTS from '../constants/constants';

class Results extends Component {

  handleClick = (event) => {
    this.props.sortMovies(event.target.id);
  };

  render() {
    return (
      <div className="bar grey lighten-1">
        <h6 id="moviesFound"
          className="white-text text-darken-2 grey lighten-1 filter-bar left">{this.props.items + " " + CONSTANTS.MOVIES_FOUND}</h6>
        <button id="RATING" onClick={this.handleClick}
          className="white-text text-darken-2 grey lighten-1 filter-bar right button-as-text">{CONSTANTS.RATING}</button>
        <button id="DATE" onClick={this.handleClick}
          className="white-text text-darken-2 grey lighten-1 filter-bar right button-as-text">{CONSTANTS.RELEASE}</button>
        <h6 className="white-text text-darken-2 grey lighten-1 filter-bar right">{CONSTANTS.SORT}</h6>
      </div>
    )
  }
}

Results.propTypes = {
  items: PropTypes.number,
  sorty: PropTypes.string,
  sortMovies: PropTypes.func
};

const mapDispatchToProps = {
  sortMovies
};

export { Results };
export default connect(null, mapDispatchToProps)(Results);
