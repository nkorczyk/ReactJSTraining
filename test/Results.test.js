import React from 'react';
import { mount } from 'enzyme';
import Results from "../src/components/Results";

describe('should render Footer component', () => {
  const element = mount(<Results items={10} />);
  test('Snapshot test with default props', () => {
    expect(element).toMatchSnapshot();
  });
});

describe('should render Footer component with 10 items', () => {
  const element = mount(<Results items={10} />);
  expect(element.prop('items')).toEqual(10);
});
