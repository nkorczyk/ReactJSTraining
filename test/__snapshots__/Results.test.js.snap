// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer component Snapshot test with default props 1`] = `
<Results
  items={10}
>
  <div
    className="bar grey lighten-1"
  >
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar left"
    >
      10 movies found
    </h6>
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar right"
    >
      rating
    </h6>
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar right"
    >
      release date
    </h6>
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar right"
    >
      Sort by
    </h6>
  </div>
</Results>
`;
